### PR TITLE
Project updated to play 2.4.0

### PR DESCRIPTION
**Pull request by kindoblue**
I am new to scala so I was not able to have the cache working, then I stripped it out :-)

### CODE REVIEW

File: project/activator-sbt-idea-shim.sbt [hunk 1]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("com.github.mpeltonen" % "sbt-idea" % "1.5.2")
```
Comment (jamesward) [inline project/activator-sbt-idea-shim.sbt]: Shouldn't need this plugin with the latest Scala IntelliJ plugin.

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.4.0")
 
-// The Typesafe repository 
-resolvers += "Typesafe repository" at "http://repo.typesafe.com/typesafe/releases/"
+// web plugins
 
-// Use the Play sbt plugin for Play projects
-addSbtPlugin("play" % "sbt-plugin" % "2.1.2")
\ No newline at end of file
+addSbtPlugin("com.typesafe.sbt" % "sbt-coffeescript" % "1.0.0")
```
Comment (jamesward) [inline project/plugins.sbt]: Shouldn't need any of the sbt-web plugins with this kind of app.

**Comment by jamesward** [comment]
Thanks for the update!  Since we really need the cache to stay under API request limits, lets figure out how to add that back.  Here are the docs: https://www.playframework.com/documentation/2.4.x/ScalaCache

Let me know when you've updated this PR with those changes.

**Comment by kindoblue** [comment]
I will take a look at the ScalaCache. And... I need to update the tests (don't tell anyone :-)

**Comment by kindoblue** [comment]
Hi,

I've tried to use the cache but if chained with the FailFast and WaitOneMinute actions, it doesn't compile.

In this way, code compiles but semantically is not what we want

``` scala
class Search @Inject()(cached: Cached) extends Controller {

def tweets(query: String) = {
    cached(query) {
      FailFast {
        WaitOneMinute {
          Action.async {
            Logger.info(s"Cache miss for $query")
            try {
              Twitter.bearerToken.flatMap { bearerToken =>
                Twitter.fetchTweets(bearerToken, query).map { response =>
                  Ok(response.json)
                }
              }
            } catch {
              case illegalArgumentException: IllegalArgumentException =>
                Logger.error("Twitter Bearer Token is missing", illegalArgumentException)
                Future(InternalServerError("Error talking to Twitter"))
            }
          }
        }
      }
    }
}
```

If I nest the cached action within FailFast and WaitOneMinute, there is a error probably due to type inference

```
[info] Compiling 1 Scala source to /Users/ice/Development/twitter-search-proxy/target/scala-2.11/classes...
[error] /Users/ice/Development/twitter-search-proxy/app/controllers/Search.scala:18: type mismatch;
[error]  found   : play.api.mvc.EssentialAction
[error]  required: play.api.mvc.Action[?]
```

Any idea?

**Comment by jamesward** [comment]
This turned out to be hard with the new `Cached` api.  I ended up bypassing it and going directly to the `CacheAPI`.  Here is the result: https://github.com/jamesward/twitter-search-proxy/blob/2.4.x/app/controllers/Search.scala

The `2.4.x` branch now contains your changes.  Let me know if you think that looks ok.  And if so, I'll merge to master.

**Comment by kindoblue** [comment]
It seems to work. Thanks for spending some time on the issue. Apparently I undertook a task bigger than me, given my current (poor) knowledge of Scala.

**Comment by jamesward** [comment]
I've made a bunch of changes to move everything to DI.  Check it out and let me know what you think.

**Comment by kindoblue** [comment]
It works great and DI is a nice addition. 👍

**Pull request merged**